### DISCUSSION TITLE
How to create upload Image section in which if image is selected and click submit, it should redirect to next page

### DISCUSSION THREAD

**Discussion by Likhin-HS** [community/community#146756] [Programming Help]
### Body

if the image is selected, the line 'no image selected' should disappear and if we click submit it should redirect to next page

// Handle the gallery button click (open gallery)
galleryButton.addEventListener('click', () => {
  modal.style.display = 'none'; // Close the modal

  // Create gallery upload section
  gallerySection.innerHTML = `
    <h3>Select an Image from Gallery</h3>
    <input type="file" id="gallery-input" accept="image/*">
    <div id="image-preview">
      <p>No image selected</p>
    </div>
    <button id="submit-gallery">Submit Image</button>
    <button id="back-to-upload-gallery">Back</button>
  `;
  document.body.appendChild(gallerySection);

  const galleryInput = document.getElementById('gallery-input');
  const imagePreview = document.getElementById('image-preview');
  const submitButton = document.getElementById('submit-gallery');
  const backButtonGallery = document.getElementById('back-to-upload-gallery');

  // Handle file input for image selection
  galleryInput.addEventListener('change', (e) => {
    const file = e.target.files[0];
    if (file) {
      const reader = new FileReader();
      reader.onload = function(event) {
        imagePreview.innerHTML = `<img src="${event.target.result}" alt="Selected Image" style="max-width: 100%; height: auto;">`;
        submitButton.style.display = 'block'; // Show the submit button
      };
      reader.readAsDataURL(file); // Read the file as Data URL for preview
    }
  });

  // Handle submit button click
  submitButton.addEventListener('click', () => {
    const file = galleryInput.files[0];
    if (!file) {
      alert("Please upload an image before submitting.");
    } else {
      window.location.href = 'photo-description.html';
    }
  });

  // Back to upload modal
  backButtonGallery.addEventListener('click', () => {
    gallerySection.remove(); // Remove gallery section
    modal.style.display = 'flex'; // Show the modal
  });

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by cachedcake** [comment]
Hello @Likhin-HS,

I think I've spotted your problem and here is the conclusion I believe would work:

```javascript
galleryButton.addEventListener('click', () => {
    modal.style.display = 'none'; 

    gallerySection.innerHTML = `
        <h3>Select an Image from Gallery</h3>
        <input type="file" id="gallery-input" accept="image/*">
        <div id="image-preview">
            <p id="no-image-text">No image selected</p>
        </div>
        <button id="submit-gallery" style="display: none;">Submit Image</button>
        <button id="back-to-upload-gallery">Back</button>
    `;
    document.body.appendChild(gallerySection);

    const galleryInput = document.getElementById('gallery-input');
    const imagePreview = document.getElementById('image-preview');
    const noImageText = document.getElementById('no-image-text');
    const submitButton = document.getElementById('submit-gallery');
    const backButtonGallery = document.getElementById('back-to-upload-gallery');

    galleryInput.addEventListener('change', (e) => {
        const file = e.target.files[0];
        if (file) {
            const reader = new FileReader();
            reader.onload = function(event) {
                noImageText.style.display = 'none';
                imagePreview.innerHTML = `<img src="${event.target.result}" alt="Selected Image" style="max-width: 100%; height: auto;">`;
                submitButton.style.display = 'block'; 
            };
            reader.readAsDataURL(file);
        }
    });


    submitButton.addEventListener('click', () => {
        const file = galleryInput.files[0];
        if (!file) {
            alert("Please upload an image before submitting.");
        } else {
            window.location.href = 'photo-description.html';
        }
    });


    backButtonGallery.addEventListener('click', () => {
        gallerySection.remove();
        modal.style.display = 'flex'; 
    });
});
```

What I changed:

- The `innerHTML` content now properly uses backticks (`` ` ``) for multi-line strings.
- Added an `id` to the `<p>` tag for easier targeting, and toggled its visibility when an image is selected.
- Ensured it is hidden initially and displayed only when an image is selected.


Let me know if this helps!

**Reply by mecodeatlas** [reply]
Hey @Likhin-HS! Did @cakecable's response help you with your question? :)
